presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -562,6 +583,26 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
         return client;
     }
 
+    private EncryptionMaterialsProvider createEncryptionMaterialsProvider(Configuration hadoopConfig)
+    {
+        EncryptionMaterialsProvider emp = null;
+        String empClassName = hadoopConfig.get(S3_ENCRYPTION_MATERIALS_PROVIDER);
```

Comment:
Yes — we implement an encryption materials provider that connects to our internal KMS, it reads initialization config from the hadoop config passed in.  This works if you place your EMP jars in the $PRESTO_HOME/plugin/hive-hadoop2 directory.

Thanks,

-nate

Nate Sammons
Principal Architect
[cid:image002.png@01CFDD78.1CD32F50]

From: David Phillips <notifications@github.com<mailto:notifications@github.com>>
Reply-To: facebook/presto <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, October 27, 2015 at 10:15 AM
To: facebook/presto <presto@noreply.github.com<mailto:presto@noreply.github.com>>
Cc: Nate Sammons <nate.sammons@nasdaq.com<mailto:nate.sammons@nasdaq.com>>
Subject: Re: [presto] Add support for S3 EncryptionMaterialsProvider to PrestoS3FileSystem (#3802)


In presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java<https://github.com/facebook/presto/pull/3802#discussion_r43144785>:

> @@ -562,6 +583,26 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
>          return client;
>      }
>
> +    private EncryptionMaterialsProvider createEncryptionMaterialsProvider(Configuration hadoopConfig)
> +    {
> +        EncryptionMaterialsProvider emp = null;
> +        String empClassName = hadoopConfig.get(S3_ENCRYPTION_MATERIALS_PROVIDER);


Do you have a custom implementation of EncryptionMaterialsProvider?

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/presto/pull/3802/files#r43144785>.
***********************************************************
CONFIDENTIALITY NOTICE: This e-mail and any attachments are for the exclusive and confidential use of the intended recipient and may constitute non-public information. If you received this e-mail in error, disclosing, copying, distributing or taking any action in reliance of this e-mail is strictly prohibited and may be unlawful. Instead, please notify us immediately by return e-mail and promptly delete this message and its attachments from your computer system. We do not waive any work product or other applicable legal privilege(s) by the transmission of this message.
***********************************************************